src/services/BookingService.ts

```diff
@@ -119,13 +121,48 @@ class BookingService {
     return booking;
   }
 
-  public async deleteById(id: number): Promise<void> {
+  public async bookingBelongsToTheUser(
+    bookingId: number, userId: number,
+  ): Promise<boolean> {
+    const repo = this.getRepository();
+    const booking = await repo.findOneBy({ id: bookingId });
+
+    console.log("Booking:");
+    console.log(booking);
+
+    if (Number(booking.user) === userId) {
+      return true;
+    }
+
+    return false;
+  }
+
+  public async deleteById(
+    bookingId: number, userId: number, userRole: string,
+  ): Promise<void> {
+    const isBookingBelongToTheUser = await this.bookingBelongsToTheUser(
+      bookingId, userId,
+    );
+
+    console.log("isBookingBelongToTheUser:");
+    console.log(isBookingBelongToTheUser);
+
+    const isAllowedUser = userRole === UserRole.ADMIN
+    || userRole === UserRole.CONTRACTOR && isBookingBelongToTheUser;
+
+    console.log("isAllowedUser");
+    console.log(isAllowedUser);
+
+    if (!isAllowedUser) {
+      throw new NotAuthorizedException();
```

Comment:
As a rule of thumb, if we are throwing a `NotAuthorizedException` in the service, we are probably doing something wrong... This should be thrown way before getting here. We need to handle this in the authorization middleware, otherwise we'll end up with authorization logic scattered around the whole application & specially the business layer.